pkg/providers/tinkerbell/stack/stack.go

```diff
@@ -119,6 +121,12 @@ func WithLoadBalancerEnabled(enabled bool) InstallOption {
 	}
 }
 
+// UpdateStackInstallerNoProxy is for workload cluster upgrade, we have to pass
+// controlPlaneEndpoint IP of managemement cluster if proxy is configured.
+func (s *Installer) UpdateStackInstallerNoProxy(IP string) {
+	s.proxyConfig.NoProxy = append(s.proxyConfig.NoProxy, IP)
```

Comment:
Can't we pass the additional IP in `NewInstaller()` instead of setting in after it's initialized?

Reply:
We can't get management cluster's control plane endpoint IP [where](https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/tinkerbell.go#LL147C45-L147C45) we initialize the stackInstaller since we don't have to context there.